File: 2025-redistribute-characters-to-make-all-strings-equal/redistribute-characters-to-make-all-strings-equal.py
class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        freq = defaultdict(int)

        for w in words:
            for c in w:
                freq[c] += 1

        for c in freq:
            if freq[c] % len(words):
                return False           
        return True

File: 2025-redistribute-characters-to-make-all-strings-equal/redistribute-characters-to-make-all-strings-equal.cpp
class Solution {
public:
    bool makeEqual(vector<string>& words) {
        int n = words.size();
        unordered_map<char,int> freq;
        for(auto &w : words){
            for(auto &c : w){
                freq[c]++;
            }
        }
        for(auto &[k, v] : freq){
            if(v % n != 0) return false;
        }
        return true;
    }
};